internal/server/models/pagination.go

```diff
@@ -0,0 +1,26 @@
+package models
+
+import (
+	"math"
+)
+
+// Internal Pagination Data
+type Pagination struct {
+	Page  int
+	Limit int
+	Sort  string
+
+	TotalCount int
+	TotalPages int
+}
+
+func (p *Pagination) SetCount(count int64) {
```

Comment:
Similar to above comment – having to "calculate" state can be a tough developer experience using this struct